src/osd/osd_types.h

```diff
@@ -5062,7 +5062,8 @@ class pg_missing_set : public pg_missing_const_i {
   void got(const hobject_t& oid, eversion_t v) {
     std::map<hobject_t, item>::iterator p = missing.find(oid);
     ceph_assert(p != missing.end());
-    ceph_assert(p->second.need <= v || p->second.is_delete());
+    ceph_assert(p->second.need <= v);
+    ceph_assert(p->second.is_delete());
```

Comment:
Hi @pponnuvel, this seems to change the logic of the assert. This approach would make sense if the line was:
```
ceph_assert(p->second.need <= v && p->second.is_delete());
```

But the `||` logic indicates that just one of the conditions is required for the assert to pass. If the current assert fails, we know that neither condition was met.

I will also update the tracker issue to get more clarity about what additional debug information is needed.